revocations.c

```diff
@@ -487,19 +486,14 @@ hibagrl_check(const struct hibagrl *grl, u_int64_t serial, u_int32_t grant_idx)
 			// We only break the loop when we found a matching
 			// serial which index we store in cur.
 			break;
-		} else if (cur_serial > serial) {
-			max = cur;
-			cur = min + ((max - min) / 2);
-		} else if (cur_serial < serial) {
-			min = cur;
-			cur = min + ((max - min) / 2) + 1;
 		}
 
-		// There is nothing left between min and max. We can return OK.
-		if (max == min) {
-			debug2("hibagrl_check: not present in GRL: 0x%" PRIx64, serial);
-			return HIBA_OK;
+		if (cur_serial > serial) {
+			max = cur - 1;
+		} else if (cur_serial < serial) {
+			min = cur + 1;
 		}
```

Comment:
This could be simply an else-block, because we exhausted the alternatives above. For symmetry reasons, I'd put the == case in the else, though.

Reply:
Done, I kept the else if condition in clear though, as it helps readability IMO